skimage/data/generate_shapes.py

```diff
@@ -0,0 +1,328 @@
+        When a shape cannot be fit into the image with the given starting
+        coordinates (x_0, y_0). This usually means the image dimensions are too
+        small or shape dimensions too large.
+
+    Returns
+    -------
+    label : Label
+        A tuple specifying the category of the shape, as well as its x1, x2, y1
+        and y2 bounding box coordinates.
+    indices: 2-D array
+        A mask of indices that the shape fills.
+    """
+    if shape.min_size == 1 or shape.max_size == 1:
+        raise ValueError('dimension must be > 1 for circles')
+    available_side = min(image.ncols - point.column, point.row + 1,
+                         shape.max_size)
+    if available_side < shape.min_size:
+        raise ArithmeticError('cannot fit shape to image')
+    side = random.randint(shape.min_size, available_side + 1)
+    triangle_height = int(np.ceil(np.sqrt(3 / 4.0) * side))
+    triangle = draw.polygon([
+        point.row,
+        point.row - triangle_height,
+        point.row,
+    ], [
+        point.column,
+        point.column + side // 2,
+        point.column + side,
+    ])
+    label = Label('triangle', point.column, point.column + side,
+                  point.row - triangle_height, point.row)
+
+    return triangle, label
+
+
+# Allows lookup by key as well as random selection.
+SHAPE_GENERATORS = dict(
+    rectangle=_generate_rectangle_mask,
+    circle=_generate_circle_mask,
+    triangle=_generate_triangle_mask)
+SHAPE_CHOICES = list(SHAPE_GENERATORS.values())
+
+
+def _generate_random_color(gray, min_pixel_intensity, random):
+    """Generates a random color array.
+
+    Parameters
+    ----------
+    gray : bool
+        If `True`, the color will be a scalar, else a 3-element array.
+    min_pixel_intensity : [0-255] int
+        The lower bound for the pixel values.
+    random : np.random.RandomState
+        The random state to use for random sampling.
+
+    Raises
+    ------
+    ValueError
+        When the min_pixel_intensity is not in the interval [0, 255].
+
+    Returns
+    -------
+    color : scalar or array
+        If gray is True, a single random scalar in the range of
+        [min_pixel_intensity, 255], else an array of three elements, each in
+        the range of [min_pixel_intensity, 255].
+
+    """
+    if not (0 <= min_pixel_intensity <= 255):
+        raise ValueError('Minimum intensity must be in interval [0, 255]')
+    if gray:
+        return random.randint(min_pixel_intensity, 256)
+    return random.randint(min_pixel_intensity, 256, size=3)
+
+
+def generate_shapes(image_shape,
+                    max_shapes,
+                    min_shapes=1,
+                    min_size=2,
+                    max_size=None,
+                    gray=False,
+                    shape=None,
+                    min_pixel_intensity=0,
+                    allow_overlap=False,
+                    number_of_attemps_to_fit_shape=100,
+                    random_seed=None):
```

Comment:
I suggest to reformat this list of parameters as compact as possible, such that it still fits on the 80 line column. One potentially useful additional argument would be `allow_partial` to allow shapes that are partially inside and outside of the image. Not strictly necessary though.

Reply:
Ok sure. Do you have any suggestions what parameters we could remove from the interface?